src/components/recordedit/multi-form-input.tsx

```diff
@@ -0,0 +1,424 @@
+            LogActions.SET_ALL_CANCEL,
+            cm.logStackPathChild,
+            cm.logStackNode,
+            undefined,
+            cm.column.reference ? cm.column.reference : undefined
+        );
+
+        toggleActiveSelectAll(columnModelIndex);
+    };
+
+    // Call back for select all checkbox to toggle all forms as selected and unselected
+    const onSelectChange = () => {
+        setShowTooltip(false);
+        if (!selectAll) {
+            setActiveForm(forms);
+        } else {
+            setActiveForm([]);
+        }
+        setSelectAll(!selectAll);
+    };
+
+    /**
+     * The callback used by functions above to set the values of the row.
+     * if clearValue is true, it will use emtpy value, otherwise it will copy the multi-form-input input value
+     */
+    const setValueForAllInputs = (clearValue?: boolean) => {
+        const cm = columnModels[columnModelIndex];
+
+        activeForms?.forEach((formValue: number) => {
+            // ignore the ones that cannot be updated
+            if (appMode === appModes.EDIT && canUpdateValues && !canUpdateValues[`${formValue}-${cm.column.name}`]) {
+                return;
+            }
+            reset(
+                copyOrClearValue(
+                    cm,
+                    getValues(),
+                    foreignKeyData.current,
+                    formValue,
+                    SELECT_ALL_INPUT_FORM_VALUE,
+                    clearValue
+                )
+            );
+        });
+    };
+
+    /** 
+     * This is to set the width of text area as the width of multi-form-input-row. We have to involve javascript as 
+     * the immidiate parent centre-align we cant set a width to it as 100%. So we have to involve JS to set the width of textarea
+     * to the next immediate parent width which is multi-form-input-row. 
+     * We choose 1800 as it matches with css rule given in multi-form-input-row and multi-form-input. Beyond 1800 we are 
+     * setting a width for textarea as 1200 and making it centre aligned. We choose 1200 as we dont want input to span
+     * across container for higher resolutions(i.e beyond 1800)
+    */ 
+    const updateTextareaWidth = () => {
+        const textarea = document.querySelector('.select-some-textarea') as HTMLElement;
+        const nonScrollableDiv = document.querySelector('.multi-form-input-row') as HTMLElement;
+        if (textarea) {
+            if (window.innerWidth < 1800) {
+                const newContainerWidth = nonScrollableDiv.offsetWidth;
+                textarea.style.width = `${newContainerWidth}px`;
+            } else {
+                textarea.style.width = '1200px';
+            }
+        }
+    };
+    // -------------------------- render logic ---------------------- //
+
+    const columnModel = columnModels[columnModelIndex];
+    const colName = columnModel.column.name;
+    const inputName = `${SELECT_ALL_INPUT_FORM_VALUE}-${colName}`;
+
+    const btnClass = `${makeSafeIdAttr(
```

Comment:
Revert this back to be just one line. The formatter that you'be used have broken this into three for no reason.